syntax = "proto3";

package data;

message Step {
  int32 step = 1;
  repeated Particle particles = 2;
  double static_energy = 3;
  double dynamic_energy = 4;
  double static_energy_variance = 5;
  double dynamic_energy_variance = 6;
  double x_value = 7;
  double v_value = 8;

  message Particle {
    int32 id = 1;
    double x = 2;
    double y = 3;
    repeated Kparam kparams = 4;
    message Kparam {
      int32 id = 1;
      float val = 2;
    }
  }
}

message Steps {
  repeated Step steps = 1;
}
